lib/id3c/api/datastore.py

```diff
@@ -178,6 +178,39 @@ def store_fhir(session: DatabaseSession, document: str) -> None:
         except (DataError, IntegrityError) as error:
             raise BadRequestDatabaseError(error) from None
 
+@export
+@catch_permission_denied
+def verify_barcode_use_list(session: DatabaseSession, barcode_use_list: str) -> dict:
+    """
+    Check the given *barcode_use_list* containing objects with ``barcode`` and ``use``
+    keys and values to verify that each barcode exists in the backing database and that the
+    given use matches the stored use.
+
+    Returns a list of objects in the same order as the input, with each object including the
+    ``barcode`` (string) being verified, ``barcode_found`` (boolean) indicating whether the
+    given barcode exists, and ``use_match`` (boolean) indicating whether the given use matches
+    the stored use. The ``use_match`` value will be `null` if the barcode does not exist.
+    """
+    barcodes = [bu['barcode'] for bu in barcode_use_list]
+    
+    barcode_use_records = session.fetch_all("""
+        select identifier.barcode, identifier_set.use from warehouse.identifier 
+        join warehouse.identifier_set using (identifier_set_id)
+        where barcode = ANY(%s)""", (barcodes,))
+
+    barcode_use_dict = {row[0]: row[1] for row in barcode_use_records}
+
+    result = []
+    for bu in barcode_use_list:
+        found = bu['barcode'] in barcode_use_dict
```

Comment:
Oh, ditto this line too, actually, with `in`. Might be easier to just keep this matching in the database and produce more of the returned data structure there.

Reply:
Refactored to match as `citext` in SQL.